lms/djangoapps/shoppingcart/tests/test_views.py

```diff
@@ -387,6 +389,23 @@ def test_course_does_not_exist_in_cart_against_valid_coupon(self):
         self.assertEqual(resp.status_code, 404)
         self.assertIn("Discount does not exist against code '{0}'.".format(self.coupon_code), resp.content)
 
+    def test_inactive_registration_code_returns_error(self):
+        """
+        test to redeem inactive registration code and
+        it returns an error.
+        """
+        course_key = self.course_key.to_deprecated_string()
+        self.add_reg_code(course_key, is_valid=False)
+        self.add_course_to_user_cart(self.course_key)
+
+        # now apply the inactive registration code
+        # it will raise an exception
+        resp = self.client.post(reverse('shoppingcart.views.use_code'), {'code': self.reg_code})
+        self.assertEqual(resp.status_code, 400)
+        self.assertIn(
+            "The registration code '{registration_code}' is not longer valid.".format(
```

Comment:
This registration code ({registration_code}) is no longer valid.

(Change the single quotation marks to parentheses and change "not" to "no")